src/pages/templates/navigation/navbar/withDarkModeSwitcher.tsx

```diff
@@ -0,0 +1,91 @@
+import { ReactNode } from 'react';
+import {
+  Box,
+  Flex,
+  Avatar,
+  Link,
+  Button,
+  Menu,
+  MenuButton,
+  MenuList,
+  MenuItem,
+  MenuDivider,
+  useDisclosure,
+  useColorModeValue,
+  Stack,
+  useColorMode,
+  Center,
+} from '@chakra-ui/react';
+import { MoonIcon, SunIcon } from '@chakra-ui/icons';
+
+const NavLink = ({ children }: { children: ReactNode }) => (
+  <Link
+    px={2}
+    py={1}
+    rounded={'md'}
+    _hover={{
+      textDecoration: 'none',
+      bg: useColorModeValue('gray.200', 'gray.700'),
+    }}
+    href={'#'}>
+    {children}
+  </Link>
+);
+
+export default function Nav() {
+  const { colorMode, toggleColorMode } = useColorMode();
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  return (
+    <>
+      <Box bg={useColorModeValue('gray.100', 'gray.900')} px={4}>
+        <Flex h={16} alignItems={'center'} justifyContent={'space-between'}>
+          <Box>Logo</Box>
+
+          <Flex alignItems={'center'}>
+            <Button onClick={toggleColorMode}>
+              {colorMode === 'light' ? <MoonIcon /> : <SunIcon />}
+            </Button>
+            &nbsp; &nbsp; &nbsp; &nbsp;
```

Comment:
What is this? This should not be necessary.

Reply:
The four NBSPs are for the spacing between the theme switcher and the user Icon, they're too close if not for the blank spaces

Reply:
You should not use space for that. You can use the `Stack` component to define spacing between elements.
https://chakra-ui.com/docs/layout/stack

Reply:
Am on mobile, will fix the next time I'm on my computer